### PR TITLE
Fix incorrect relative/absolute URL usages

### PR DESCRIPTION
**Pull request by wxiaoguang**
Add two "HTMLURL" methods for PackageDescriptor. And rename "FullWebLink" to "VersionWebLink"

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Wait for more reviews from #20280

```
Registry: pd.VersionWebLink(), // TODO: is it right to use VersionWebLink here?
```

It really looks strange to me.

**Comment by silverwind** [comment]
It begs the question, what is ha HTML URL? I find that a highly confusing term we should avoid.

**Comment by wxiaoguang** [comment]
> What is a HTML URL? I find that a confusing term we should remove.

Absolute URL with full scheme / host like "https://gitea/user/repo/foo/bar". It has been widely used in code.

Maybe you could be interested in #22831 and #22839

**Comment by KN4CK3R** [comment]
> What is a HTML URL? I find that a confusing term we should remove.

Agree, I would prefer something like `AbsoluteURL` and `RelativeURL`,

**Comment by wxiaoguang** [comment]
> > What is a HTML URL? I find that a confusing term we should remove.
> 
> Agree, I would prefer something like `AbsoluteURL` and `RelativeURL`,

I have the same feeling long time ago. But GitHub also uses "html_url" https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28 

And this PR is just a quick fix. If most people would like to rename, I think it needs a plan & design  ahead.

**Comment by wxiaoguang** [comment]
TLDR: at the moment, in most cases, "Link" means relative URL (`issue.Link()`), while "HTMLURL" means absolute URL (`issue.HTMLURL()`)

**Comment by wxiaoguang** [comment]
> Wait for more reviews from #20280
> 
> ```
> Registry: pd.VersionWebLink(), // TODO: is it right to use VersionWebLink here?
> ```
> 
> It really looks strange to me.

If there is no more idea about whether it is right, I will just keep it as the old code (a relative link), and merge. New fixes could be done in the future.

**Comment by KN4CK3R** [comment]
> > It really looks strange to me.
> 
> If there is no more idea about whether it is right, I will just keep it as the old code (a relative link), and merge. New fixes could be done in the future.

Currently looking for an answer. 👍

**Comment by KN4CK3R** [comment]
Should be an absolute URL but not to a package but to the registry itself.
Same as used here:
https://github.com/go-gitea/gitea/blob/70c126e6184872a6ac63cae2f327fc745b25d1d7/routers/api/packages/npm/npm.go#L74-L77

**Comment by wxiaoguang** [comment]
Thank you very much. Done in 0625d18abb8ac0db3972703b77705d3f7590bb81

**Review by KN4CK3R — APPROVED** [review]
